### DISCUSSION TITLE
Progress bars incorrect in log

### DISCUSSION THREAD

**Discussion by mark-wd** [vladmandic/sdnext#1078] [Q&A] [answered]
### Issue Description

For several versions now, the progress bars never really complete in my logs.

![Screenshot 2023-05-19 033427](https://github.com/vladmandic/automatic/assets/31883671/33344938-dbe9-49ed-ba38-66134e78dfae)

Reproductible or an issue in my install?

### Version Platform Description

2023-05-18
hash: 527dc0ee

**Comment by vladmandic** [comment]
What do you mean? It shows 40/40?
Or are you talking about model loading? It shows total model size, but how model loading works is different as it may a) load just part it needs or b) run with memory mapped model so load is actually a no-op as load happens as weights are moved, not loaded.
In either case, this is a valid question, but is it an issue?

**Comment by vladmandic** [comment]
converting to a discussion.

**Comment by mark-wd** [comment]
I had not thought of that distinction. Yes, the model loading bar. It's confusing UX that they look partially filled. In other software that are not for generative AI this means incorrect loading.

**Reply by vladmandic** [reply]
yeah, if you want, you can make it a full bar - settings -> stable diffusion -> When loading models attempt stream loading optimized for slow or network storage

that disables memory mapping during load and loads file as a one long stream.

that's great if you're loading over network share or foreign filesystem.
but if you're loading from local ssd, memory mapping is much faster.

(btw, that option does not exist in a1111, i did quite a lot of work on that) :)

**Comment by mark-wd** [comment]
Interesting... maybe it would be useful to print in the log something like "loaded via (method x)". One of the worst things to me about the main branch was always the lack of UX, so it's something that I am always looking towards in this branch, which does much better in that.

**Comment by vladmandic** [comment, marked as answer]
not a problem, i'll add it to tomorrows release - example `shared.log.debug('Model weights loading: type=checkpoint mode=direct')`
debug messages are visible when running with `--debug` flag, i don't want to overload standard output.
